mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -326,4 +322,80 @@ def GENX_AtomicRMWOp : GENX_Op<"atomic.rmw", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Joint Matrix operations
+//===----------------------------------------------------------------------===//
+
+def GENX_IsJointMatrixType :
+  CPred<"$_self.isa<::mlir::GENX::JointMatrixType>()">;
+
+def GENX_AnyJointMatrix : DialectType<GENX_Dialect, GENX_IsJointMatrixType,
+                                "any GENX joint matrix type">;
+
+def ScopeKind : I32EnumAttr<"Scope", "GENX Scope", 
+  [
+    I32EnumAttrCase<"Device", 0>,
+    I32EnumAttrCase<"Workgroup", 1>,
+    I32EnumAttrCase<"Subgroup", 2>
+  ]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def ScopeAttr : EnumAttr<GENX_Dialect, ScopeKind, "scope"> {
+  let assemblyFormat = "`<` $value `>`";
+}
+
+def MatrixLayoutKind : I32EnumAttr<"MatrixLayout", "GENX Matrix Layout", 
+  [
+    I32EnumAttrCase<"ColumnMajor", 0>,
+    I32EnumAttrCase<"RowMajor", 1>,
+    I32EnumAttrCase<"PackedA", 2>,
+    I32EnumAttrCase<"PackedB", 3>
+  ]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def MatrixLayoutAttr : EnumAttr<GENX_Dialect, MatrixLayoutKind, "layout"> {
+  let assemblyFormat = "`<` $value `>`";
+}
+
+def MemoryAccessKind : I32BitEnumAttr<"MemoryAccess", "GENX Memory Access", 
+  [
+    I32BitEnumAttrCaseBit<"None", 0>,
+    I32BitEnumAttrCaseBit<"Volatile", 1>,
+    I32BitEnumAttrCaseBit<"Aligned", 2>,
+    I32BitEnumAttrCaseBit<"NonTemporal", 3>
+  ]> {
+  let genSpecializedAttr = 0;
+  let cppNamespace = "::mlir::GENX";
+}
+def MemoryAccessAttr : EnumAttr<GENX_Dialect, MemoryAccessKind, "memory_access"> {
+  let assemblyFormat = "`<` $value `>`";
+}
+
+def GENX_MatrixLoadOp : GENX_Op<"matrix.load">,
+  Results<(outs GENX_AnyJointMatrix:$res)>,
+  Arguments<(ins Arg<LLVM_AnyPointer,"", [MemRead]>:$ptr,
+             Index:$stride, MatrixLayoutAttr:$layout, ScopeAttr:$scope,
+             OptionalAttr<MemoryAccessAttr>:$memory_access)> {
+  let summary = "Load a matrix through a pointer";
+
+  string baseDescription = [{
+    The `genx.matrix.load` operation performs a load of the matrix at the memory 
+    location specified by the pointer argument %ptr. The stride argument specifies 
+    the number of matrix elements in memory between the start of successive rows/columns 
```

Comment:
missing an `and` somewhere in the sentence? between the start of successive rows/columns in the result and what?